### PR TITLE
Add geometry_type_to_name function

### PR DESCRIPTION
**Pull request by ttencate**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

### CODE REVIEW

File: src/vector/geometry.rs [hunk 1]
```diff
@@ -437,11 +437,16 @@ impl PartialEq for Geometry {
 
 impl Eq for Geometry {}
 
+pub fn geometry_type_to_name(ty: OGRwkbGeometryType::Type) -> String {
+    let rv = unsafe { gdal_sys::OGRGeometryTypeToName(ty) };
```
Comment (lnicola) [inline src/vector/geometry.rs:441]: Might be worth adding a null check here, even if it's just an assert.

Comment (lnicola) [reply src/vector/geometry.rs:441]: Actually, `OGRwkbGeometryType::Type` is an integer, so it's easy to call this with an invalid value. I'd be inclined to return `Option<String>`.

Comment (ttencate) [reply src/vector/geometry.rs:441]: Result<String> then? It's more consistent with other APIs in the crate.

Comment (ttencate) [reply src/vector/geometry.rs:441]: https://gdal.org/api/vector_c_api.html#_CPPv421OGRGeometryTypeToName18OGRwkbGeometryType says "NULL on failure". Let's do an assertion then.

Got to run now, will do this next week (or feel free to do it yourself if you have time).

Comment (ttencate) [reply src/vector/geometry.rs:441]: Done now, and locked down with an extra test.

**Comment by lnicola** [comment]
Oof, if you're still around, can you rebase to fix the changelog conflict?

**Comment by ttencate** [comment]
Btw how would you feel about having this awkward `OGRwkbGeometryType` hammered into a proper Rust `enum` called `GeometryType`? Just a matter of someone not having done it yet, or are there objections?

**Comment by ttencate** [comment]
I will run Clippy and rustfmt before pushing next time. I will run Clippy and rustfmt before pushing next time. I will run Clippy and rustfmt before pushing next time.

**Comment by lnicola** [comment]
> Btw how would you feel about having this awkward OGRwkbGeometryType hammered into a proper Rust enum called GeometryType? Just a matter of someone not having done it yet, or are there objections?

Might be a good idea, but I'm not sure how extensible it's supposed to be (take a look at the [values defined](https://gdal.org/doxygen/ogr__core_8h.html#a800236a0d460ef66e687b7b65610f12a) in GDAL).

**Pull request merged**